packages/flutter/test/painting/text_painter_test.dart

```diff
@@ -832,6 +832,125 @@ void main() {
     )!;
     expect(caretHeight, 50.0);
   }, skip: isBrowser); // https://github.com/flutter/flutter/issues/56308
+
+  group('TextPainter line-height', () {
+    test('half-leading', (){
+      const TextStyle style = TextStyle(
+        height: 20,
+        fontSize: 1,
+        leadingDistribution: TextLeadingDistribution.even,
+      );
+
+      final TextPainter painter = TextPainter()
+        ..textDirection = TextDirection.ltr
+        ..text = const TextSpan(text: 'A', style: style)
+        ..layout();
+
+      final Rect glyphBox = painter.getBoxesForSelection(
+        const TextSelection(baseOffset: 0, extentOffset: 1),
+      ).first.toRect();
+
+      final RelativeRect insets = RelativeRect.fromSize(glyphBox, painter.size);
+      // The glyph box is centered.
+      expect(insets.top, insets.bottom);
+      // The glyph box is exactly 1 logical pixel high.
+      expect(insets.top, (20 - 1) / 2);
+    });
+
+    test('half-leading with small height', (){
+      const TextStyle style = TextStyle(
+        height: 0.1,
+        fontSize: 10,
+        leadingDistribution: TextLeadingDistribution.even,
+      );
+
+      final TextPainter painter = TextPainter()
+        ..textDirection = TextDirection.ltr
+        ..text = const TextSpan(text: 'A', style: style)
+        ..layout();
+
+      final Rect glyphBox = painter.getBoxesForSelection(
+        const TextSelection(baseOffset: 0, extentOffset: 1),
+      ).first.toRect();
+
+      final RelativeRect insets = RelativeRect.fromSize(glyphBox, painter.size);
+      // The glyph box is still centered.
+      expect(insets.top, insets.bottom);
+      // The glyph box is exactly 10 logical pixel high. Negative leading.
+      expect(insets.top, (1 - 10) / 2);
```

Comment:
Where does the 1 come from?  I was expecting a 0.1 from the height.

Reply:
Added to the comment that the height multiplier does not scale the glyph only the line-height.